1.9.0 aarch64 with gcc14 build fail

### Problem Description

release 1.9.0 aarch64 with gcc14 build fail

### Activity Log, Crash Log or any other details

```text
D_FILE_OFFSET_BITS=64 -DHAS_STRERROR_R -I./libhb/ -I./contrib/include -c ../libhb/work.c -o libhb/work.o
  : In file included from ../libhb/comb_detect.c:142:
  : ../libhb/templates/comb_detect_template.c: In function вЂdetect_gamma_combed_segment_8вЂ™:
  : ../libhb/templates/comb_detect_template.c:273:27: error: passing argument 1 of вЂvst1q_u32вЂ™ from incompatible pointer type [-Wincompatible-pointer-types]
  :   273 |                 vst1q_u32(&mask32, mask_vec);
  :       |                           ^~~~~~~
  :       |                           |
  :       |                           uint32_t (*)[4] {aka unsigned int (*)[4]}
  : In file included from ../libhb/decomb.c:202:
  : ../libhb/templates/decomb_template.c: In function вЂblend_filter_line_8вЂ™:
  : ../libhb/templates/decomb_template.c:266:19: error: passing argument 1 of вЂvst1q_u16вЂ™ from incompatible pointer type [-Wincompatible-pointer-types]
  :   266 |         vst1q_u16(&cr_table_vec, result_u16);
  :       |                   ^~~~~~~~~~~~~
  :       |                   |
  :       |                   uint16_t (*)[8] {aka short unsigned int (*)[8]}
  : In file included from ../libhb/comb_detect.c:35:
  : /usr/lib64/gcc/aarch64-rosa-linux-gnu/14.2.0/include/arm_neon.h:22174:22: note: expected вЂuint32_t *вЂ™ {aka вЂunsigned int *вЂ™} but argument is of type вЂuint32_t (*)[4]вЂ™ {aka вЂunsigned int (*)[4]вЂ™}
  : 22174 | vst1q_u32 (uint32_t *__a, uint32x4_t __b)
  :       |            ~~~~~~~~~~^~~
  : In file included from ../libhb/templates/decomb_template.c:19:
  : /usr/lib64/gcc/aarch64-rosa-linux-gnu/14.2.0/include/arm_neon.h:22167:22: note: expected вЂuint16_t *вЂ™ {aka вЂshort unsigned int *вЂ™} but argument is of type вЂuint16_t (*)[8]вЂ™ {aka вЂshort unsigned int (*)[8]вЂ™}
  : 22167 | vst1q_u16 (uint16_t *__a, uint16x8_t __b)
  :       |            ~~~~~~~~~~^~~
  : ../libhb/templates/decomb_template.c:246:15: warning: unused variable вЂfilter_norm_vecвЂ™ [-Wunused-variable]
  :   246 |     int16x8_t filter_norm_vec = vdupq_n_s16(-filter->normalize);
  :       |               ^~~~~~~~~~~~~~~
  : In file included from ../libhb/comb_detect.c:146:
  : ../libhb/templates/comb_detect_template.c: In function вЂdetect_gamma_combed_segment_16вЂ™:
  : ../libhb/templates/comb_detect_template.c:273:27: error: passing argument 1 of вЂvst1q_u32вЂ™ from incompatible pointer type [-Wincompatible-pointer-types]
  :   273 |                 vst1q_u32(&mask32, mask_vec);
  :       |                           ^~~~~~~
  :       |                           |
  :       |                           uint32_t (*)[4] {aka unsigned int (*)[4]}
  : /usr/lib64/gcc/aarch64-rosa-linux-gnu/14.2.0/include/arm_neon.h:22174:22: note: expected вЂuint32_t *вЂ™ {aka вЂunsigned int *вЂ™} but argument is of type вЂuint32_t (*)[4]вЂ™ {aka вЂunsigned int (*)[4]вЂ™}
  : 22174 | vst1q_u32 (uint32_t *__a, uint32x4_t __b)
  :       |            ~~~~~~~~~~^~~
  : ../libhb/templates/comb_detect_template.c: In function вЂdetect_combed_segment_16вЂ™:
  : ../libhb/templates/comb_detect_template.c:553:35: error: passing argument 1 of вЂvst1q_u32вЂ™ from incompatible pointer type [-Wincompatible-pointer-types]
  :   553 |                         vst1q_u32(&mask32, mask_vec);
  :       |                                   ^~~~~~~
  :       |                                   |
  :       |                                   uint32_t (*)[4] {aka unsigned int (*)[4]}
  : /usr/lib64/gcc/aarch64-rosa-linux-gnu/14.2.0/include/arm_neon.h:22174:22: note: expected вЂuint32_t *вЂ™ {aka вЂunsigned int *вЂ™} but argument is of type вЂuint32_t (*)[4]вЂ™ {aka вЂunsigned int (*)[4]вЂ™}
  : 22174 | vst1q_u32 (uint32_t *__a, uint32x4_t __b)
  :       |            ~~~~~~~~~~^~~
  : ../libhb/templates/comb_detect_template.c:571:35: error: passing argument 1 of вЂvst1q_u32вЂ™ from incompatible pointer type [-Wincompatible-pointer-types]
  :   571 |                         vst1q_u32(&mask32, mask_vec);
  :       |                                   ^~~~~~~
  :       |                                   |
  :       |                                   uint32_t (*)[4] {aka unsigned int (*)[4]}
  : /usr/lib64/gcc/aarch64-rosa-linux-gnu/14.2.0/include/arm_neon.h:22174:22: note: expected вЂuint32_t *вЂ™ {aka вЂunsigned int *вЂ™} but argument is of type вЂuint32_t (*)[4]вЂ™ {aka вЂunsigned int (*)[4]вЂ™}
  : 22174 | vst1q_u32 (uint32_t *__a, uint32x4_t __b)
  :       |            ~~~~~~~~~~^~~
  : ../libhb/templates/comb_detect_template.c:588:35: error: passing argument 1 of вЂvst1q_u32вЂ™ from incompatible pointer type [-Wincompatible-pointer-types]
  :   588 |                         vst1q_u32(&mask32, mask_vec);
  :       |                                   ^~~~~~~
  :       |                                   |
  :       |                                   uint32_t (*)[4] {aka unsigned int (*)[4]}
  : /usr/lib64/gcc/aarch64-rosa-linux-gnu/14.2.0/include/arm_neon.h:22174:22: note: expected вЂuint32_t *вЂ™ {aka вЂunsigned int *вЂ™} but argument is of type вЂuint32_t (*)[4]вЂ™ {aka вЂunsigned int (*)[4]вЂ™}
  : 22174 | vst1q_u32 (uint32_t *__a, uint32x4_t __b)
  :       |            ~~~~~~~~~~^~~
  : In file included from ../libhb/decomb.c:206:
  : ../libhb/templates/decomb_template.c: In function вЂblend_filter_line_16вЂ™:
  : ../libhb/templates/decomb_template.c:184:19: error: passing argument 1 of вЂvst1q_u32вЂ™ from incompatible pointer type [-Wincompatible-pointer-types]
  :   184 |         vst1q_u32(&cr_table_vec, result_u32);
  :       |                   ^~~~~~~~~~~~~
  :       |                   |
  :       |                   uint32_t (*)[4] {aka unsigned int (*)[4]}
  : /usr/lib64/gcc/aarch64-rosa-linux-gnu/14.2.0/include/arm_neon.h:22174:22: note: expected вЂuint32_t *вЂ™ {aka вЂunsigned int *вЂ™} but argument is of type вЂuint32_t (*)[4]вЂ™ {aka вЂunsigned int (*)[4]вЂ™}
  : 22174 | vst1q_u32 (uint32_t *__a, uint32x4_t __b)
  :       |            ~~~~~~~~~~^~~
  : ../libhb/templates/decomb_template.c:164:15: warning: unused variable вЂfilter_norm_vecвЂ™ [-Wunused-variable]
  :   164 |     int32x4_t filter_norm_vec = vdupq_n_s32(-filter->normalize);
  :       |               ^~~~~~~~~~~~~~~
  : gmake: *** [../libhb/module.rules:12: libhb/comb_detect.o] Error 1
```
```


### What Operating System are you running?

rosa13

### What version of HandBrake are you running?

1.9.0

### Where did you download HandBrake from?

github

--- a/libhb/templates/comb_detect_template.c
+++ b/libhb/templates/comb_detect_template.c
@@ -270,7 +270,7 @@ static void FUNC(detect_gamma_combed_segment)(hb_filter_private_t *pv,
                 mask_vec = vandq_u32(mask_vec, condition);
                 mask_vec = vandq_u32(mask_vec, v_one);
 
-                vst1q_u32(&mask32, mask_vec);
+                vst1q_u32(mask32, mask_vec);
 
                 mask[0] = mask32[0];
                 mask[1] = mask32[1];
@@ -550,7 +550,7 @@ static void FUNC(detect_combed_segment)(hb_filter_private_t *pv,
                         mask_vec = vandq_u32(mask_vec, condition);
                         mask_vec = vandq_u32(mask_vec, v_one);
 
-                        vst1q_u32(&mask32, mask_vec);
+                        vst1q_u32(mask32, mask_vec);
 
                         mask[0] = mask32[0];
                         mask[1] = mask32[1];
@@ -568,7 +568,7 @@ static void FUNC(detect_combed_segment)(hb_filter_private_t *pv,
                         mask_vec = vandq_u32(mask_vec, condition);
                         mask_vec = vandq_u32(mask_vec, v_one);
 
-                        vst1q_u32(&mask32, mask_vec);
+                        vst1q_u32(mask32, mask_vec);
 
                         mask[0] = mask32[0];
                         mask[1] = mask32[1];
@@ -585,7 +585,7 @@ static void FUNC(detect_combed_segment)(hb_filter_private_t *pv,
                         mask_vec = vandq_u32(mask_vec, condition);
                         mask_vec = vandq_u32(mask_vec, v_one);
 
-                        vst1q_u32(&mask32, mask_vec);
+                        vst1q_u32(mask32, mask_vec);
 
                         mask[0] = mask32[0];
                         mask[1] = mask32[1];

--- a/libhb/templates/decomb_template.c
+++ b/libhb/templates/decomb_template.c
@@ -181,7 +181,7 @@ static void FUNC(blend_filter_line)(const filter_param_t *filter,
         result = vshrq_n_s32(result, 3);
 
         uint32x4_t result_u32 = vreinterpretq_u32_s32(vaddq_s32(result, offset));
-        vst1q_u32(&cr_table_vec, result_u32);
+        vst1q_u32(cr_table_vec, result_u32);
         dst[x+0] = crop_table[cr_table_vec[0]];
         dst[x+1] = crop_table[cr_table_vec[1]];
         dst[x+2] = crop_table[cr_table_vec[2]];
@@ -263,7 +263,7 @@ static void FUNC(blend_filter_line)(const filter_param_t *filter,
         result = vshrq_n_s16(result, 3);
 
         uint16x8_t result_u16 = vreinterpretq_u16_s16(vaddq_s16(result, offset));
-        vst1q_u16(&cr_table_vec, result_u16);
+        vst1q_u16(cr_table_vec, result_u16);
         dst[x+0] = crop_table[cr_table_vec[0]];
         dst[x+1] = crop_table[cr_table_vec[1]];
         dst[x+2] = crop_table[cr_table_vec[2]];

